packages/flutter_tools/test/general.shard/vmservice_test.dart

```diff
@@ -166,7 +166,7 @@ void main() {
 
   testUsingContext('VMService returns correct FlutterVersion', () async {
     final MockVMService mockVMService = MockVMService();
-    setUpVmService(
+    await setUpVmService(
```

Comment:
Are there any additional checks to add here around the async `registerService` calls being done after this await completes?

Reply:
Sorry, not sure what you mean exactly

Reply:
If the `Future`s returned by the `registerService` calls weren't being `await`ed, but this PR now `await`s them, then the behavior is a bit different, and I'm wondering if any additional checks can be put after this call to make sure it doesn't regress. If the premise is wrong, and the behavior isn't any different, then this comment can be ignored.

Reply:
Generally we assume that all of the futures will complete successfully, that isn't quite right. I updated this method so that it catches RPC errors and tool exits